lib/content-services/src/lib/content-metadata/components/content-metadata/content-metadata.component.spec.ts

```diff
@@ -412,14 +433,190 @@ describe('ContentMetadataComponent', () => {
 
             fixture.detectChanges();
             tick(100);
-            clickOnSave();
+            clickOnGeneralInfoSave();
 
             tick(100);
             expect(component.node).toEqual(expectedNode);
             expect(updateService.updateNodeAspect).toHaveBeenCalledWith(expectedNode);
         }));
     });
 
+    describe('toggleEdit', () => {
+        let mockEvent: MouseEvent;
+        let showErrorSpy: jasmine.Spy;
+        const mockGroup: CardViewGroup = {
+            editable: false,
+            expanded: false,
+            title: '',
+            properties: []
+        };
+        beforeEach(() => {
+            mockEvent = new MouseEvent('click');
+            component.editableGroup = mockGroup;
+            showErrorSpy = spyOn(notificationService, 'showError').and.stub();
```

Comment:
Why are you reapplying the spy before each test?

Reply:
We need these preconditions for each test case.
The reason is to add 'showErrorSpy' when we update the data from any panel and click on another panel edit button we are showing the snackbar message. So it is applicable for all edit icons.